packages/frontend-api/src/Model/Resolver/Category/CategoryResolver.php

```diff
@@ -18,26 +22,103 @@ class CategoryResolver implements ResolverInterface, AliasedInterface
+     */
+    protected ?FriendlyUrlFacade $friendlyUrlFacade;
+
     /**
      * @param \Shopsys\FrameworkBundle\Model\Category\CategoryFacade $categoryFacade
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain|null $domain
+     * @param \Shopsys\FrontendApiBundle\Model\FriendlyUrl\FriendlyUrlFacade|null $friendlyUrlFacade
      */
     public function __construct(
-        CategoryFacade $categoryFacade
+        CategoryFacade $categoryFacade,
+        ?Domain $domain = null,
+        ?FriendlyUrlFacade $friendlyUrlFacade
     ) {
         $this->categoryFacade = $categoryFacade;
+        $this->domain = $domain;
+        $this->friendlyUrlFacade = $friendlyUrlFacade;
     }
 
     /**
-     * @param string $uuid
+     * @required
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @internal This function will be replaced by constructor injection in next major
+     */
+    public function setDomain(Domain $domain): void
+    {
+        if ($this->domain !== null && $this->domain !== $domain) {
+            throw new BadMethodCallException(sprintf(
+                'Method "%s" has been already called and cannot be called multiple times.',
+                __METHOD__
+            ));
+        }
+        if ($this->domain !== null) {
+            return;
+        }
+        @trigger_error(
+            sprintf(
+                'The %s() method is deprecated and will be removed in the next major. Use the constructor injection instead.',
+                __METHOD__
+            ),
+            E_USER_DEPRECATED
+        );
+        $this->domain = $domain;
+    }
+
+    /**
+     * @required
+     * @param \Shopsys\FrontendApiBundle\Model\FriendlyUrl\FriendlyUrlFacade $friendlyUrlFacade
+     * @internal This function will be replaced by constructor injection in next major
+     */
+    public function setFriendlyUrlFacade(FriendlyUrlFacade $friendlyUrlFacade): void
+    {
+        if ($this->friendlyUrlFacade !== null && $this->friendlyUrlFacade !== $friendlyUrlFacade) {
+            throw new BadMethodCallException(sprintf(
+                'Method "%s" has been already called and cannot be called multiple times.',
+                __METHOD__
+            ));
+        }
+        if ($this->friendlyUrlFacade !== null) {
+            return;
+        }
+        @trigger_error(
+            sprintf(
+                'The %s() method is deprecated and will be removed in the next major. Use the constructor injection instead.',
+                __METHOD__
+            ),
+            E_USER_DEPRECATED
+        );
+        $this->friendlyUrlFacade = $friendlyUrlFacade;
+    }
+
+    /**
+     * @param string|null $uuid
+     * @param string|null $urlSlug
      * @return \Shopsys\FrameworkBundle\Model\Category\Category
      */
-    public function resolver(string $uuid): Category
+    public function resolver(?string $uuid = null, ?string $urlSlug = null): Category
```

Comment:
This is a BC break – I tested extending and I get error
```
Compile Error: Declaration of 
Shopsys\FrontendApiBundle\Model\Resolver\Category\Category2Resolver::resolver(string $uuid): Shopsys\FrameworkBundle\Model\Category\Category
must be compatible with 
Shopsys\FrontendApiBundle\Model\Resolver\Category\CategoryResolver::resolver(?string $uuid = null, ?string $urlSlug = null): Shopsys\FrameworkBundle\Model\Category\Category
```
(see https://symfony.com/doc/current/contributing/code/bc.html#changing-classes)

I know we talked about it and I'm not sure how we tested it, maybe it's guarded by Symfony and not by PHP itself